### DISCUSSION TITLE
Getting/Decoding previewData from the client side?

### DISCUSSION THREAD

**Discussion by josias-r** [vercel/next.js#12064] [Help] [answered]
The new [preview API](https://nextjs.org/docs/advanced-features/preview-mode) allows you to set some preview cookies.

Currently, when I'm calling my enable-preview endpoint, I'm returning the same data as JSON that I set as my previewData. This allows me to store those values inside the local storage for my client.

Now since the cookies might time out after a while, my client still has the local storage values set.
Is there some way of accessing the cookies directly from the client-side, which would mean I wouldn't have to store anything inside local storage.

Currently, the cookies seem to be encrypted and I don't know how to access the data stored inside it.

**Comment by timneutkens** [comment, marked as answer]
It's not possible to read the cookie client-side as that would be insecure.

**Reply by josias-r** [reply]
From the docs I can read:
>You can pass an object to setPreviewData and have it be available in getStaticProps. However, because the data will be stored in a cookie, there’s a size limitation. Currently, preview data is limited to 2KB.

Since it says "currently", is there a chance this logic might change or will previewData never be available from the client-side?

**Reply by Timer** [reply]
The "currently" is referring to the 2KB cap. The data will never be made available to client-side JS except through `getStaticProps`.

**Reply by rikkit** [reply]
Is it possible to read the cookie server-side, in a middleware?
